functions/hold_peak.h

```diff
@@ -0,0 +1,55 @@
+#ifndef FUNCTIONS_HOLD_PEAK_H
+#define FUNCTIONS_HOLD_PEAK_H
+
+// Usage: HoldPeakF<F, HOLD_MILLIS, SPEED>
+// Holds Peak value of F for HOLD_MILLIS.
+// then transitions down over SPEED to current F
+// F, HOLD_MILLIS and SPEED: FUNCTION
+// return value: FUNCTION, same for all LEDs
+
+class BladeBase;
+
+template<class F, class HOLD_MILLIS, class SPEED>
+class HoldPeakF {
+public:
+  void run(BladeBase* blade) {
+    f_.run(blade);
+    speed_.run(blade);
+    hold_time_millis.run(blade);
+    current = f_.getInteger(0);
+    hold_millis = hold_time_millis.getInteger(0);
+    uint32_t now = micros();
+    uint64_t delta = now - last_micros;
+    last_micros = now;
+    if (current > value_) {
+      value_ = current;
+      last_peak = millis();
+    }
+    if (value_ > current && millis() - last_peak > hold_millis) {
+      if (delta > 1000000) delta = 1;
+      delta *= speed_.getInteger(0);
+      delta /= 1000000;
+      if (delta > (value_ - current)) {
```

Comment:
the purpose of this if statement is to make sure that value_ doesn't go below current.
However, it would be simpler to just do value_ -= delta, and then do the check on line 24 *after*, which also makes sure that value_ doesn't go below current. Like this:

   if (millis() - last_peak > hold_millis) {
       if (delta > 1000000) delta = 1;
      delta *= speed_.getInteger(0);
      delta /= 1000000;
      value_ -= delta;
  }
  if (current > value_) {
      value_ = current;
      last_peak = millis();
  }

Reply:
thanks, works as expected, I have updated